apple/Sources/FerrostarCarPlayUI/CarPlayModels/CPManeuver.swift

```diff
@@ -3,6 +3,30 @@ import FerrostarCore
 import FerrostarCoreFFI
 import FerrostarSwiftUI
 
+private let InstructionKey = "com.stadiamaps.ferrostar.instruction"
+
+extension CPManeuver {
+    private var userDictionary: [String: Any]? {
+        get {
+            userInfo as? [String: Any] ?? [:]
+        }
+        set {
+            userInfo = newValue
+        }
+    }
+
+    public var visualInstruction: VisualInstruction? {
+        get {
+            userDictionary?[InstructionKey] as? VisualInstruction
+        }
+        set {
+            var info = userDictionary ?? [:]
+            info[InstructionKey] = newValue
+            userDictionary = info
```

Comment:
Is there a better way to do this nowadays? This looks like a data race waiting to happen.

Reply:
This value is set once, read more often. The property is not a part of the initializer, unfortunately.

Reply:
This type doesn't have a specific initializer, so it's a bunch of properties to set just once and get thereafter.